modules/social_features/social_core/src/ContentTranslationConfigOverrideBase.php

```diff
@@ -56,10 +43,9 @@ public function loadOverrides($names) {
 
     // This setting can't be changed in an override because that would create
     // and endless loop in trying to apply the override.
-    $settings = \Drupal::configFactory()->getEditable('social_content_translation.settings');
-    $is_enabled = $settings->getOriginal($this->getModule(), FALSE);
+    $is_content_translations_enabled = \Drupal::moduleHandler()->moduleExists('social_content_translation');
 
-    if ($is_enabled) {
+    if ($is_content_translations_enabled) {
```

Comment:
@nechai I think we can move this whole config to the `social_content_translation` module. Then we don't need this extra check and spaghetti code. 

@Kingdutch do you agree with that? Or was there some specific reason to have it in `social_core` rather than in `social_content_translation` ?

Reply:
@ribel I think we still need this check otherwise config overrides will be loaded even if the module "social_content_translate" be disabled (PSR-4 class autoloading) - class loading(detecting) doesn't require module enabling....

Reply:
Seems like I was wrong on that point - services can't load the class from the disabled module. 

So this expression `use Drupal\social_content_translation\ContentTranslationConfigOverrideBase` will raise the error if module `social_content_translation` is disabled (and it's disabled by default in distro).
I tried to declare our config overrides services with ServiceProvider but got the same issue....

We need to leave the base class `ContentTranslationConfigOverrideBase` and live with the "spaghetti code" or create another approach to resolve the problem 🤔